pkg/kubelet/kubelet_pods_test.go

```diff
@@ -264,41 +313,81 @@ fe00::2	ip6-allrouters
 				{IP: "123.45.67.89", Hostnames: []string{"foo", "bar", "baz"}},
 				{IP: "456.78.90.123", Hostnames: []string{"park", "doo", "boo"}},
 			},
-			expectedContent: `# Kubernetes-managed hosts file.
-127.0.0.1	localhost
-::1	localhost ip6-localhost ip6-loopback
-fe00::0	ip6-localnet
-fe00::0	ip6-mcastprefix
-fe00::1	ip6-allnodes
-fe00::2	ip6-allrouters
-203.0.113.1	podFoo.domainFoo	podFoo
+			isSupportIPv6: true,
+			expectedContent: `
+				# Kubernetes-managed hosts file.
+				127.0.0.1	localhost
+				::1	localhost ip6-localhost ip6-loopback
+				fe00::0	ip6-localnet
+				fe00::0	ip6-mcastprefix
+				fe00::1	ip6-allnodes
+				fe00::2	ip6-allrouters
+				203.0.113.1	podFoo.domainFoo	podFoo
+
+				# Entries added by HostAliases.
+				123.45.67.89	foo	bar	baz
+				456.78.90.123	park	doo	boo
+				`,
+		},
+		{
+			hostIPs:        []string{"203.0.113.1"},
+			hostName:       "podFoo",
+			hostDomainName: "domainFoo",
+			hostAliases: []v1.HostAlias{
+				{IP: "123.45.67.89", Hostnames: []string{"foo", "bar", "baz"}},
+				{IP: "456.78.90.123", Hostnames: []string{"park", "doo", "boo"}},
+			},
+			isSupportIPv6: false,
+			expectedContent: `
+				# Kubernetes-managed hosts file.
+				127.0.0.1	localhost
+				203.0.113.1	podFoo.domainFoo	podFoo
 
-# Entries added by HostAliases.
-123.45.67.89	foo	bar	baz
-456.78.90.123	park	doo	boo
-`,
+				# Entries added by HostAliases.
+				123.45.67.89	foo	bar	baz
+				456.78.90.123	park	doo	boo
+				`,
 		},
 		{
 			hostIPs:        []string{"203.0.113.1", "fd00::6"},
 			hostName:       "podFoo",
 			hostDomainName: "domainFoo",
 			hostAliases:    []v1.HostAlias{},
-			expectedContent: `# Kubernetes-managed hosts file.
-127.0.0.1	localhost
-::1	localhost ip6-localhost ip6-loopback
-fe00::0	ip6-localnet
-fe00::0	ip6-mcastprefix
-fe00::1	ip6-allnodes
-fe00::2	ip6-allrouters
-203.0.113.1	podFoo.domainFoo	podFoo
-fd00::6	podFoo.domainFoo	podFoo
-`,
+			isSupportIPv6:  true,
+			expectedContent: `
+				# Kubernetes-managed hosts file.
+				127.0.0.1	localhost
+				::1	localhost ip6-localhost ip6-loopback
+				fe00::0	ip6-localnet
+				fe00::0	ip6-mcastprefix
+				fe00::1	ip6-allnodes
+				fe00::2	ip6-allrouters
+				203.0.113.1	podFoo.domainFoo	podFoo
+				fd00::6	podFoo.domainFoo	podFoo
+				`,
+		},
+		{
+			hostIPs:        []string{"203.0.113.1", "fd00::6"},
+			hostName:       "podFoo",
+			hostDomainName: "domainFoo",
+			hostAliases:    []v1.HostAlias{},
+			isSupportIPv6:  false,
+			expectedContent: `
+		# Kubernetes-managed hosts file.
```

Comment:
I feel like this could indent more so it looks like it's "under" `expectedContent` ?  The others seem right